content/posts/github-workflow-guide-for-modules.md

```diff
@@ -0,0 +1,370 @@
+
+The final step is another official action provided by GitHub [that stores a build artifact](https://github.com/marketplace/actions/upload-a-build-artifact) from one job so that it can be used by subsequent jobs in the same workflow. It needs to be given a `name` (think a cache key) and `path` to the file to store as its arguments, which are typically passed into actions using the `with` key. The `if` key allows a step to be run conditionally. In this case, a variable inside the runner is evaluated using `${{ ... }}`. We're checking whether the thing running this workflow is [a tool called `act`](https://github.com/nektos/act) (not the YAPC website). It allows you to test workflows locally on your computer without pushing lots of commits or force-pushing changes every time you make a change to the workflow file. It is handy, but you don't need this condition unless you plan to use `act`.
+
+This brings us to the end of the build job. We now have a distribution stored and ready to be used in the subsequent jobs.
+
+The coverage job
+---
+
+```yaml
+  coverage-job:
+    needs: build-job
+    runs-on: ubuntu-20.04
+    container:
+      image: perldocker/perl-tester:5.34
+    steps:
+      - uses: actions/checkout@v3 # codecov wants to be inside a Git repository
+      - uses: actions/download-artifact@v3
+        with:
+          name: build_dir
+          path: .
+      - name: Install deps and test
+        run: cpan-install-dist-deps && test-dist
+        env:
+          CODECOV_TOKEN: ${{secrets.CODECOV_TOKEN}}
+```
+
+Although we're going to run a whole array of tests on different platforms later on in the workflow file, we don't want to collect test coverage data for all of them. Instead, we have defined a dedicated test run just to collect coverage data using [codecov.io](https://about.codecov.io/), which is free for Open Source projects. You need to create an account, which you can conveniently do with your GitHub login, and you also need to sync your repositories and enable them inside of codecov. You will be given a token, which you need to store in your GitHub repository's settings under Secrets inside Actions. In this example, we've named it `CODECOV_TOKEN`.
+
+The job again runs on the Perl docker container, and we've opted for Perl 5.34. Because we want to run this after the build job has finished, we set the `needs` key to the name of that job.
```

Comment:
```suggestion
The job again runs on the Perl Docker container, and we've opted for Perl 5.34. Because we want to run this after the build job has finished, we set the `needs` key to the name of that job.
```